packages/node/src/integrations/tracing/vercelai/index.ts

```diff
@@ -3,152 +3,153 @@ import { SEMANTIC_ATTRIBUTE_SENTRY_OP, defineIntegration, spanToJSON } from '@se
 import type { IntegrationFn } from '@sentry/core';
 import { generateInstrumentOnce } from '../../../otel/instrument';
 import { addOriginToSpan } from '../../../utils/addOriginToSpan';
-import { SentryVercelAiInstrumentation, sentryVercelAiPatched } from './instrumentation';
+import { SentryVercelAiInstrumentation } from './instrumentation';
 
 const INTEGRATION_NAME = 'VercelAI';
 
 export const instrumentVercelAi = generateInstrumentOnce(INTEGRATION_NAME, () => new SentryVercelAiInstrumentation({}));
 
 const _vercelAIIntegration = (() => {
+  let instrumentation: undefined | SentryVercelAiInstrumentation;
+
   return {
     name: INTEGRATION_NAME,
     setupOnce() {
-      instrumentVercelAi();
-    },
-    processEvent(event) {
-      if (event.type === 'transaction' && event.spans?.length) {
-        for (const span of event.spans) {
-          const { data: attributes, description: name } = span;
-
-          if (!name || span.origin !== 'auto.vercelai.otel') {
-            continue;
-          }
-
-          if (attributes['ai.usage.completionTokens'] != undefined) {
-            attributes['ai.completion_tokens.used'] = attributes['ai.usage.completionTokens'];
-          }
-          if (attributes['ai.usage.promptTokens'] != undefined) {
-            attributes['ai.prompt_tokens.used'] = attributes['ai.usage.promptTokens'];
-          }
-          if (
-            typeof attributes['ai.usage.completionTokens'] == 'number' &&
-            typeof attributes['ai.usage.promptTokens'] == 'number'
-          ) {
-            attributes['ai.total_tokens.used'] =
-              attributes['ai.usage.completionTokens'] + attributes['ai.usage.promptTokens'];
-          }
-        }
-      }
-
-      return event;
+      instrumentation = instrumentVercelAi();
```

Comment:
We should convert this to use `instrumentWhenWrapped`.

Reply:
I was thinking about this - technically, this implementation is _better_ and safer, as it does not rely on patching the method etc. Since we control the instrumentation class here 🤔